public/app/features/datasources/components/EditDataSource.tsx

```diff
@@ -128,11 +129,14 @@ export function EditDataSourceView({
     e.preventDefault();
     trackDsConfigClicked('save_and_test');
 
+    const eventData = parseEventDataFromDataSource(dataSource);
+
     try {
       await onUpdate({ ...dataSource });
       trackDsConfigUpdated({ item: 'success' });
-      appEvents.publish(new DataSourceUpdatedSuccessfully());
+      appEvents.publish(new DataSourceUpdatedSuccessfully(eventData));
```

Comment:
Yes, anyone would be able to listen to these events (at least with the current version of the appEvents).

Reply:
Yes, that's my understanding as well. I think any subscriber that is set up while the page is loaded can access this information, so it's important that we keep this to be non-sensitive data.

Reply:
@sarahzinger we should consider than from a privacy and security perspective we should simply not expose anything from a plugin to another regardless of what we consider sensitive data or not. 

Something you can do is to keep the state of the configuration in your plugin datasource and listen to the event to later flush the data you already have but broadcasting the plugin configuration all over the application is dangerous.

Reply:
Really great call outs @academo! Let me know what you think of this solution:
- publish events without any associated data
- then in the config component that has the state information we're looking for we subscribe and report there.

I think I was stuck thinking app event subscriptions had to happen in module.tsx since that's what I had seen us do elsewhere with other events, but in this case there's really no reason to!